### ISSUE TITLE
continue pretrain OOM 问题？

### ISSUE THREAD

**Issue by ReadyTeresa** [xiaomi-research/gemmax#6]
continue pretrain 已经将batch_size调小了，还是OOM
参数如下：
    --model_name_or_path ${pretrained_model} \
    --stage pt \
    --do_train true \
    --seed 42 \
    --preprocessing_num_workers 1 \
    --finetuning_type full \
    --template empty \
    --flash_attn disabled \
    --dataset_dir ${dataset_dir} \
    --dataset ${dataset_name} \
    --cutoff_len 512 \
    --learning_rate 2e-05 \
    --num_train_epochs 1 \
    --max_samples 100000000000 \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 24 \
    --lr_scheduler_type cosine \
    --warmup_ratio 0.01 \
    --weight_decay 0.01 \
    --max_grad_norm 1.0 \
    --optim adamw_torch \
    --bf16 true \
    --logging_strategy steps \
    --logging_steps 5 \
    --save_steps 636 \
    --packing true \
    --plot_loss true \
    --ddp_timeout 180000000 \
    --include_num_input_tokens_seen true \
    --output_dir ${outputs} \
    --deepspeed examples/deepspeed/ds_z2_config.json

**Comment by gpengzhi** [comment]
Thanks for your interest!

Could you provide more detailed information, such as your base model, training infrastructure, etc.?

**Comment by gpengzhi** [comment]
Could you also provide the training log?

**Comment by gpengzhi** [comment]
> > Could you also provide the training log?
> 
> 2025-06-27T11:16:30.314813279+08:00 [INFO|tokenization_utils_base.py:2058] 2025-06-27 11:16:30,314 >> loading file vocab.json 49 2025-06-27T11:16:30.314838072+08:00 [INFO|tokenization_utils_base.py:2058] 2025-06-27 11:16:30,314 >> loading file merges.txt 50 2025-06-27T11:16:30.314842658+08:00 [INFO|tokenization_utils_base.py:2058] 2025-06-27 11:16:30,314 >> loading file tokenizer.json 51 2025-06-27T11:16:30.314865169+08:00 [INFO|tokenization_utils_base.py:2058] 2025-06-27 11:16:30,314 >> loading file added_tokens.json 52 2025-06-27T11:16:30.314891589+08:00 [INFO|tokenization_utils_base.py:2058] 2025-06-27 11:16:30,314 >> loading file special_tokens_map.json 53 2025-06-27T11:16:30.314913281+08:00 [INFO|tokenization_utils_base.py:2058] 2025-06-27 11:16:30,314 >> loading file tokenizer_config.json 54 2025-06-27T11:16:30.314917333+08:00 [INFO|tokenization_utils_base.py:2058] 2025-06-27 11:16:30,314 >> loading file chat_template.jinja 55 2025-06-27T11:16:30.642763146+08:00 [INFO|tokenization_utils_base.py:2323] 2025-06-27 11:16:30,642 >> Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained. 56 2025-06-27T11:16:30.667364622+08:00 [WARNING|2025-06-27 11:16:30] llamafactory.data.template:148 >> `template` was not specified, try parsing the chat template from the tokenizer. 57 2025-06-27T11:16:30.680931066+08:00 [INFO|2025-06-27 11:16:30] llamafactory.data.loader:143 >> Loading dataset /home/work/data/data.json... 58 2025-06-27T11:20:01.655553281+08:00 59 Generating train split: 0 examples [00:00, ? examples/s] killed

There is no specific error message. I'm unable to determine if it's a GPU OOM issue. Is your single training dataset huge? If so, you could split it into multiple pieces.
